packages/std/src/errors.rs

```diff
@@ -81,12 +89,8 @@ pub fn invalid<T>(field: &'static str, msg: &'static str) -> StdResult<T> {
     ValidationErr { field, msg }.fail()
 }
 
-pub fn contract_err<T>(msg: &'static str) -> StdResult<T> {
-    ContractErr { msg }.fail()
-}
-
-pub fn dyn_contract_err<T>(msg: String) -> StdResult<T> {
-    DynContractErr { msg }.fail()
+pub fn dyn_contract_err<T, S: Into<String>>(msg: S) -> StdResult<T> {
```

Comment:
This does simplify from the static case, but is needless heap allocations, which is why I avoided it. (Moving &'static str to String is not free).

That said, this is only called in error cases, and not like it is executed in a loop in the success case, so shouldn't really affect contract gas costs. After a bit of reflection I am happy with the simplification. But please just call it `contract_err` now.

Reply:
I agree.

I wanted to rename to `contract_err`, but I wonder how this error type differentiates from the other ones? What does "contract error" mean? From the usage we have right now, where it is also used in the mock API, it feels more like a `general_err` or `unspecified_err`. Any thoughts that help differentiating or better describing this one?

Reply:
`generic_err`?
`untyped_err`?

Reply:
`generic_err` works well, let's take that.

Another thing with `&'static str` in errors: any idea how it is possible to deserialize JSON into `&'static str` fields? It seems to work for `NotFound` but I don't understand why.

Reply:
> Another thing with `&'static str` in errors: any idea how it is possible to deserialize JSON into `&'static str` fields? It seems to work for `NotFound` but I don't understand why.

Here is the answer to the question: bf08d8b70c444aeeae79e2951659a939abdbcd3a. Static error fields can only be deserialized from static sources.